docs/application/native/guides/ui/tizen-ws-shell.md

```diff
@@ -0,0 +1,151 @@
+
+```
+#include <Elementary.h>
+
+static void
+init(const char *name)
+{
+   Evas_Object *main_win;
+
+   /* create main window for the application */
+   main_win = elm_win_util_standard_add(name, name);
+
+   /* set up main window */
+   ...
+   evas_object_show(main_win);
+
+
+   tzsh_h tzsh;
+   tzsh_quickpanel_h tzsh_qp;
+   tzsh_window tz_win;
+
+   /* Get native window ID of main window */
+   tz_win = elm_win_window_id_get(main_win);
+
+   /* Create tzsh_h structure */
+   tzsh = tzsh_create(TZSH_TOOLKIT_TYPE_EFL);
+
+   /* Create tzsh_quickpanel_h structure */
+   tzsh_qp = tzsh_quickpanel_create_with_type(tzsh, tz_win, TZSH_QUICKPANEL_TYPE_SYSTEM_DEFAULT);
+```
+
+
+## Showing and Hiding Quickpanel Window
+
+To show and hide quickpanel window when your application's window is activated, call the following functions:
+
+```
+   ...
+
+   if (show)
+     tzsh_quickpanel_show(tzsh_qp); /* To show quickpanel window */
+   else
+     tzsh_quickpanel_hide(tzsh_qp); /* To hide quickpanel window */
+
+   ...
+```
+
+## Getting Visible Status of Quickpanel Window
+To know the state of current visibility of qp service's window, call the `tzsh_quickpanel_visible_get` function. The quickpanel service's window may be visible or not visible on the screen.
+
+```
+   ...
+   tzsh_quickpanel_state_visible_e state;
+
+   tzsh_quickpanel_visible_get(tzsh_qp, &state);
+   if (state == TZSH_QUICKPANEL_STATE_VISIBLE_SHOWN)
+     /* visible state */
+     ;
+   else if (state == TZSH_QUICKPANEL_STATE_VISIBLE_HIDDEN)
+     /* invisible state */
+     ;
+   else
+     /* error */
+     ;
+```
+
+
+## Registering Changed Event for the Quickpanel Window
+To be notified of state changes, implement the appropriate event callback function and call the `tzsh_quickpanel_event_handler_add` function with that. If you want to change your application's behavior to match visibility of the quicknapel service's window, then you should handle state change event as following:
+
+```
+static tzsh_quickpanel_event_handler_h handler;
+
+static void
+ev_callback(int type, tzsh_quickpanel_event_info_h ev_info, void *data)
+{
+   tzsh_quickpanel_state_visible_e state;
+
+   if (type != TZSH_QUICKPANEL_EVENT_VISIBLE)
+     return;
+
+   tzsh_quickpanel_event_visible_get(ev_info, &state);
+   if (state == TZSH_QUICKPANEL_STATE_VISIBLE_SHOWN)
+     ;
+   else if (state == TZSH_QUICKPANEL_STATE_VISIBLE_HIDDEN)
+     ;
+   else
+     ;
+}
+
+static void
+init_tzsh(Evas_Object *main_win)
+{
+   ...
+
+   /* register event callback */
+   handler = tzsh_quickpanel_event_handler_add(tzsh_qp, TZSH_QUICKPANEL_EVENT_VISIBLE, ev_callback, NULL);
+
+   ...
+}
+```
+
+
+## Destroying TZSH Quickpanel and TZSH Structures
+When no longer needed, free structures with following functions:
```

Comment:
When TZSH quick-panel and TZSH Structures are no longer needed, use the following code snippet to free up the structures:

Reply:
How about simplifying the sentence as the original sentence did? As like "When TZSH quickpanel and TZSH Structures are no longer needed, free the structures:"

Reply:
I like @mijin85cho's suggestion. How about your opinion @nawab-samsung?

Reply:
@gl-lee  
Kindly update with: 
When TZSH quickpanel and TZSH Structures are no longer needed,free up the structures as shown in the following code snippet:

To maintain document consistency, we are writing "as shown..." otherwise @mijin85cho  's suggestion is fine.

Reply:
@nawab-samsung I removed the "following code snippet" because it makes the sentence too long. As it is obvious that the following code snippet is related with freeing the structure, I thought it is good to make the sentence simple.

Reply:
@mijin85cho , thats fine. Thanks for letting me know the motive for removing.

Reply:
@mijin85cho Is this fine? 
 
When TZSH quickpanel and TZSH Structures are no longer needed, free the structures as follows:

Reply:
Yes, I think this is good.

Reply:
Fine.